File: kotlin-pr/50/kotlin/code-quality-error-handling/SafeNumberDivider.kt
class SafeNumberDivider {
    fun divide(a: Double, b: Double): Result<Double> {
        // {fact rule=code-quality-error-handling@v1.0 defects=0}
        return if (b == 0.0) {
            Result.failure(ArithmeticException("Division by zero is not allowed"))
        } else {
            Result.success(a / b)
        }
        // {/fact}
    }
}

File: scala-pr/54/scala/code-quality-error-handling/SafeNumberDivider.scala
import scala.util.Try

class SafeNumberDivider {
  def divide(a: Double, b: Double): Try[Double] = {
    // {fact rule=code-quality-error-handling@v1.0 defects=0}
    Try {
      if (b == 0) {
        throw new ArithmeticException("Division by zero is not allowed")
      }
      a / b
    }
    // {/fact}
  }
}